src/3d/qgspointcloudlayerchunkloader_p.cpp

```diff
@@ -42,52 +45,75 @@
 
 ///@cond PRIVATE
 
-QgsPointCloud3DGeometry::QgsPointCloud3DGeometry( Qt3DCore::QNode *parent, const QgsPointCloud3DSymbolHandler::PointData &data )
+QgsPointCloud3DGeometry::QgsPointCloud3DGeometry( Qt3DCore::QNode *parent, const QgsPointCloud3DSymbolHandler::PointData &data, QgsPointCloud3DSymbol *symbol )
   : Qt3DRender::QGeometry( parent )
   , mPositionAttribute( new Qt3DRender::QAttribute( this ) )
-  , mClassAttribute( new Qt3DRender::QAttribute( this ) )
+  , mParameterAttribute( new Qt3DRender::QAttribute( this ) )
+  , mColorAttribute( new Qt3DRender::QAttribute( this ) )
 #if QT_VERSION < QT_VERSION_CHECK(5, 10, 0)
   , mVertexBuffer( new Qt3DRender::QBuffer( Qt3DRender::QBuffer::VertexBuffer, this ) )
 #else
   , mVertexBuffer( new Qt3DRender::QBuffer( this ) )
 #endif
+  , mRenderingStyle( symbol->renderingStyle() )
 {
+  unsigned int byte_stride = mRenderingStyle == QgsPointCloud3DSymbol::RenderingStyle::RGBRendering ? 28 : 16;
```

Comment:
The byte stride could be moved to a virtual method in QgsPointCloud3DSymbol instead to keep the symbol implementations nice and self-contained.